src/sdk/pynni/nni/pbt_tuner/pbt_tuner.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+pbt_tuner.py
+"""
+
+import copy
+import logging
+import os
+import numpy as np
+
+import nni
+from nni.tuner import Tuner
+from nni.utils import NodeType, OptimizeMode, extract_scalar_reward, split_index
+
+import nni.parameter_expressions as parameter_expressions
+
+logger = logging.getLogger('pbt_tuner_AutoML')
+
+
+def json2space(x, oldy=None, name=NodeType.ROOT):
```

Comment:
please move these functions into [util.py](https://github.com/microsoft/nni/blob/master/src/sdk/pynni/nni/utils.py) and contain only one version in tuners.

Reply:
modified